src/cosmoz-treenode.ts

```diff
@@ -1,6 +1,37 @@
 import { component, html } from '@pionjs/pion';
+import { Tree, Node } from '@neovici/cosmoz-tree/cosmoz-tree';
 
-export const computePathToRender = (path, hideFromRoot, showMaxNodes) => {
+export type ComputePathTextType = {
+	ownerTree: Tree;
+	ellipsis: string;
+	pathToRender: string | undefined;
+	path: string;
+	valueProperty: string;
+	pathSeparator: string | undefined;
+};
+
+export type RenderType = {
+	title: string;
+	text: string;
+};
+
+export type TreenodeType = {
+	valueProperty: string;
+	pathSeparator: string;
+	hideFromRoot: number;
+	showMaxNodes: number;
+	keyProperty: string;
+	keyValue: string;
+	ownerTree: Tree;
+	ellipsis: string;
+	fallback: string;
+};
+
+export const computePathToRender = (
+		path?: string,
+		hideFromRoot?: number,
```

Comment:
hideFromRoot and showMaxNodes can't be optional.
path also can't be optional but you can make it string|undefined.